domain/src/main/kotlin/no/nav/su/se/bakover/domain/behandling/Behandlingsinformasjon.kt

```diff
@@ -33,48 +45,24 @@ data class Behandlingsinformasjon(
         ektefelle = b.ektefelle ?: this.ektefelle,
     )
 
-    private fun erFerdigbehandlet() =
-        listOf(
-            uførhet,
-            flyktning,
-            lovligOpphold,
-            fastOppholdINorge,
-            oppholdIUtlandet,
-            formue,
-            personligOppmøte,
-            bosituasjon,
-            ektefelle,
-        ).all { it != null && it.erGyldig() && it.erFerdigbehandlet() }
-
-    fun erInnvilget() =
-        listOf(
-            uførhet?.erVilkårOppfylt(),
-            flyktning?.erVilkårOppfylt(),
-            lovligOpphold?.erVilkårOppfylt(),
-            fastOppholdINorge?.erVilkårOppfylt(),
-            oppholdIUtlandet?.erVilkårOppfylt(),
-            formue?.erVilkårOppfylt(),
-            personligOppmøte?.erVilkårOppfylt(),
-        ).all { it ?: false }
-
-    fun erAvslag() = erFerdigbehandlet() && !erInnvilget()
+    private fun erFerdigbehandlet() = alleVilkår.all { it != null && it.erGyldig() && it.erFerdigbehandlet() }
+    fun erInnvilget() = alleVilkår.all { it?.erVilkårOppfylt() ?: false }
+    fun getAvslagsgrunn() = alleVilkår.mapNotNull { it?.avslagsgrunn() }.firstOrNull()
+    fun erAvslag(): Boolean {
+        if (uførhet != null && flyktning != null) {
+            if (uførhet.status == Uførhet.Status.VilkårIkkeOppfylt || flyktning.status == Flyktning.Status.VilkårIkkeOppfylt) {
```

Comment:
En filleting, men mulig det blir penere hvis du slår disse sammen (litt avhengig av hva logikken egentlig skal være):
```suggestion
        if (uførhet != null && uførhet.status == Uførhet.Status.VilkårIkkeOppfylt || flyktning != null && flyktning.status == Flyktning.Status.VilkårIkkeOppfylt) {
```

Kunne også gjenopptatt tanken på å ha en egen funksjon på Base som heter noe alá `erVilkårIkkeOppfylt` bare for å slippe å både nullsjekke og sjekke på statusen.